In [1]:
import pandas as pd
from zipfile import ZipFile
import yfinance as fp
import datetime

In [2]:
# with ZipFile("Data/NIFTY50_all.csv.zip","r") as ref:
#     unzip = ref.extractall("Data/Bnk_NIFTY50")
# ref.close()

In [3]:
today = datetime.date.today()
year  = today-datetime.timedelta(days=7300) #20 Years
Stocks = ['AAPL', 'MSFT', 'IBM', 'GE', 'JNJ']
data = fp.download(Stocks,start=year,end=today,group_by="ticker")
data

C:\Users\lipun\AppData\Local\Temp\ipykernel_3120\1640078772.py:4: FutureWarning: YF.download() has changed argument auto_adjust default to True
  data = fp.download(Stocks,start=year,end=today,group_by="ticker")
[*********************100%***********************]  5 of 5 completed


Ticker             JNJ                                                \
Price             Open        High         Low       Close    Volume   
Date                                                                   
2005-07-28   36.218332   36.420540   36.016120   36.358757   6391600   
2005-07-29   36.330670   36.386839   35.915012   35.926247   6143200   
2005-08-01   36.089155   36.707023   36.089155   36.229580   7348200   
2005-08-02   36.308200   36.347519   36.044201   36.302582   5554800   
2005-08-03   36.162162   36.516034   36.122843   36.437397   5090400   
...                ...         ...         ...         ...       ...   
2025-07-16  160.300003  166.119995  159.800003  164.779999  22134800   
2025-07-17  163.179993  164.699997  162.300003  162.979996  11295700   
2025-07-18  163.929993  165.100006  163.360001  163.699997   9792600   
2025-07-21  163.770004  164.919998  162.779999  164.360001   8254000   
2025-07-22  165.000000  168.059998  164.710007  167.929993   9277700   

Ticker            AAPL                                                 ...  \
Price             Open        High         Low       Close     Volume  ...   
Date                                                                   ...   
2005-07-28    1.317876    1.322385    1.301347    1.316374  251311200  ...   
2005-07-29    1.309160    1.333805    1.270090    1.281811  562080400  ...   
2005-08-01    1.279407    1.294734    1.264680    1.284817  314249600  ...   
2005-08-02    1.289024    1.307357    1.280609    1.298041  296875600  ...   
2005-08-03    1.298040    1.301647    1.285417    1.298941  258322400  ...   
...                ...         ...         ...         ...        ...  ...   
2025-07-16  210.300003  212.399994  208.639999  210.160004   47490500  ...   
2025-07-17  210.570007  211.800003  209.589996  210.020004   48068100  ...   
2025-07-18  210.869995  211.789993  209.699997  211.179993   48974600  ...   
2025-07-21  212.100006  215.779999  211.630005  212.479996   51377400  ...   
2025-07-22  213.139999  214.949997  212.229996  214.399994   46300400  ...   

Ticker              GE                                                \
Price             Open        High         Low       Close    Volume   
Date                                                                   
2005-07-28  101.783035  102.279110  101.549590  101.783035   3392115   
2005-07-29  101.549590  101.899758  100.644981  100.674164   4118009   
2005-08-01  100.878441  101.024347   99.827926   99.944649   4024132   
2005-08-02   99.944649  100.849258   99.798743   99.944649   4008274   
2005-08-03   99.973817  100.353169   99.477743   99.827911   3738014   
...                ...         ...         ...         ...       ...   
2025-07-16  264.000000  266.859985  262.029999  266.179993  11912900   
2025-07-17  269.799988  272.799988  258.160004  260.279999  16260200   
2025-07-18  265.399994  269.700012  262.559998  263.269989   8619800   
2025-07-21  263.690002  266.250000  259.579987  262.859985   5208500   
2025-07-22  260.730011  261.500000  254.660004  259.000000   4746200   

Ticker            MSFT                                                 
Price             Open        High         Low       Close     Volume  
Date                                                                   
2005-07-28   17.912401   17.981964   17.849795   17.912401   44738700  
2005-07-29   17.933271   18.086308   17.801101   17.815014   59524400  
2005-08-01   17.954135   18.121086   17.919355   18.030655   61346800  
2005-08-02   18.016746   18.712373   17.995878   18.649767  137510100  
2005-08-03   18.614986   19.081056   18.594116   18.955843  139422400  
...                ...         ...         ...         ...        ...  
2025-07-16  505.179993  506.720001  501.890015  505.619995   15154400  
2025-07-17  505.679993  513.369995  505.619995  511.700012   17503100  
2025-07-18  514.479980  514.640015  507.429993  510.049988   21209700  
2025-07-21  506.709991  5

In [4]:
new = []
for i in Stocks:
    df = data[i].copy()
    df = df.reset_index()
    df["Stocks"] = i
    new.append(df)
new_data = pd.concat(new)
new_data  

Price       Date        Open        High         Low       Close     Volume  \
0     2005-07-28    1.317876    1.322385    1.301347    1.316374  251311200   
1     2005-07-29    1.309160    1.333805    1.270090    1.281811  562080400   
2     2005-08-01    1.279407    1.294734    1.264680    1.284817  314249600   
3     2005-08-02    1.289024    1.307357    1.280609    1.298041  296875600   
4     2005-08-03    1.298040    1.301647    1.285417    1.298941  258322400   
...          ...         ...         ...         ...         ...        ...   
5022  2025-07-16  160.300003  166.119995  159.800003  164.779999   22134800   
5023  2025-07-17  163.179993  164.699997  162.300003  162.979996   11295700   
5024  2025-07-18  163.929993  165.100006  163.360001  163.699997    9792600   
5025  2025-07-21  163.770004  164.919998  162.779999  164.360001    8254000   
5026  2025-07-22  165.000000  168.059998  164.710007  167.929993    9277700   

Price Stocks  
0       AAPL  
1       AAPL  
2       AAPL  
3       AAPL  
4       AAPL  
...      ...  
5022     JNJ  
5023     JNJ  
5024     JNJ  
5025     JNJ  
5026     JNJ  

[25135 rows x 7 columns]

In [5]:
new_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 25135 entries, 0 to 5026
Data columns (total 7 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   Date    25135 non-null  datetime64[ns]
 1   Open    25135 non-null  float64       
 2   High    25135 non-null  float64       
 3   Low     25135 non-null  float64       
 4   Close   25135 non-null  float64       
 5   Volume  25135 non-null  int64         
 6   Stocks  25135 non-null  object        
dtypes: datetime64[ns](1), float64(4), int64(1), object(1)
memory usage: 1.5+ MB


In [6]:
#Saving File
new_data.to_csv("Data/DataPreparation.csv",index=False)

<h1><b>FINAL REPORT: DATA-PREPARATION</b></h1>
    <li>
        Data-Source: <a href="yfinance">Yfinance</a>
    </li>
    <li>Converted Raw Dataset to Structured Format With a New Feature as Stocks</li>
    <li><b>Total Samples:</b> 25135</li>
    <li><b>Total Independent Features: </b>"Date","Open","High","Low" ,"Volume and "Stocks"</li>
    <li><b>Dependent Feature:</b> "Close"</li>
    <center><b><li>Stocks Names</li></b></center>
    <table>
        <th>Stocks</th>
        <th>Symbols</th>
        <tr><td>AAPL</td><td>Apple</td></tr>
        <tr><td>MSFT</td><td>Microsoft</td></tr>
        <tr><td>IBM</td><td>IBM</td></tr>
        <tr><td>GE</td><td>General Electric</td></tr>
       <tr><td>JNJ</td><td>Johnson & Johnson</td></tr>
    </table>
    <center><li><b>Feature Dtypes</b></li></center>
    <table>
        <th>Features</th>
        <th>DataTypes</th>
        <tr><td>Date</td><td>datetime64</td></tr>
        <tr><td>Open</td><td>float64</td></tr>
        <tr><td>High</td><td>float64</td></tr>
        <tr><td>Low</td><td>float64</td></tr>
        <tr><td>Close</td><td>float64</td></tr>
        <tr><td>Volume</td><td>int64</td></tr>
        <tr><td>Stocks</td><td>object</td></tr>
    </table>